torch/csrc/jit/tensorexpr/eval.h

```diff
@@ -368,11 +369,21 @@ class SimpleIREvaluator : public CodeGen, public IRVisitor {
     if (lhs_v.dtype() != rhs_v.dtype()) {
       throw malformed_input("bad dtype in binary op", v);
     }
+
     IRNodeType expr_type = v->expr_type();
     if (expr_type == IRNodeType::kAnd || expr_type == IRNodeType::kOr ||
         expr_type == IRNodeType::kXor || expr_type == IRNodeType::kLshift ||
         expr_type == IRNodeType::kRshift) {
-      value_ = bitwise_binary_op(lhs_v, rhs_v, expr_type);
+      switch (lhs_v.dtype().scalar_type()) {
+#define TYPE_CASE(Type, Name)                                  \
+  case ScalarType::Name:                                       \
+    value_ = bitwise_binary_op<Type>(lhs_v, rhs_v, expr_type); \
+    break;
+        AT_FORALL_INT_TYPES(TYPE_CASE);
+#undef TYPE_CASE
+        default:
```

Comment:
Do we need to handle `ScalarType::Bool` here as well?

Reply:
You are right. According to `ir.h`, `kAnd`, `kOr`, `kXor` applies to all integral types (including `Bool`, according to `tensorexpr::is_integral()`):

```
class And : public BinaryOpNode<And> {
 public:
  And(const Expr* lhs, const Expr* rhs)
      : BinaryOpNode(lhs, rhs, IRNodeType::kAnd) {
    if (!lhs->dtype().is_integral()) {
      throw unsupported_dtype();
    }
    if (lhs->dtype() != rhs->dtype()) {
      throw malformed_input("bad dtype in And");
    }
  }
};
``` 

But for `kLshift` and `kRshift` (according to `tensorexpr/ir.h`), the supported type seems to be only `ScalarType::Int`
```
class Lshift : public BinaryOpNode<Lshift> {
 public:
  Lshift(const Expr* lhs, const Expr* rhs)
      : BinaryOpNode(lhs, rhs, IRNodeType::kLshift) {
    if (lhs->dtype().scalar_type() != ScalarType::Int) {
      throw unsupported_dtype();
    }
    if (lhs->dtype() != rhs->dtype()) {
      throw malformed_input("bad dtype in Lshift");
    }
  }
};
```

I'll make the changes accordingly.